tests/fsharp/Compiler/ErrorMessages/SuggestionsTests.fs

```diff
@@ -0,0 +1,410 @@
+            39
+            (6, 11, 6, 15)
+            "The field, constructor or member 'ello' is not defined.\nMaybe you want one of the following:\s+Hello"
+
+
+    //TODO This was marked with SCFLAGS="--vserrors" in tests/fsharpqa/Source/Warnings/env.lst file
+    [<Test>]
+    let ``Suggest Record Type for RequireQualifiedAccess Records`` () =
+        CompilerAssert.TypeCheckSingleError
+            """
+[<RequireQualifiedAccess>]
+type MyRecord = {
+    Field1: string
+    Field2: int
+}
+
+let r = { Field1 = "hallo"; Field2 = 1 }
+
+exit 0
+            """
+            FSharpErrorSeverity.Error
+            39
+            (8, 11, 8, 17)
+            "The record label 'Field1' is not defined.\nMaybe you want one of the following:\s+MyRecord.Field1"
+
+
+    ///TODO Check if the second error message is correct
+    [<Test>]
+    let ``Suggest To Use Indexer`` () =
+        CompilerAssert.TypeCheckSingleError
+            """
+let d = [1,1] |> dict
+let y = d[1]
+
+let z = d[|1|]
+
+let f() = d
+let a = (f())[1]
+
+exit 0
+            """
+            [|
+                FSharpErrorSeverity.Error, 3217, (2, 10, 2, 13), "This value is not a function and cannot be applied. Did you intend to access the indexer via d.\[index\] instead?"
+                FSharpErrorSeverity.Error, 3, (5, 9, 5, 9), "This value is not a function and cannot be applied.$"
+                FSharpErrorSeverity.Error, 3217, (8, 10, 8, 13), "This expression is not a function and cannot be applied. Did you intend to access the indexer via expr.\[index\] instead?"
+            |]
+
+
+    [<Test>]
+    let ``Suggest Type Parameters`` () =
+        CompilerAssert.TypeCheckSingleError
+            """
+[<AbstractClass>]
+type MyClass<'Bar>() =
+   abstract M<'T> : 'T -> 'T
+   abstract M2<'T> : 'T -> 'Bar
+   abstract M3<'T> : 'T -> 'B
+
+exit 0
+            """
+            FSharpErrorSeverity.Error
+            39
+            (6, 28, 6, 30)
+            "The type parameter 'B is not defined\nMaybe you want one of the following:\n'Bar\n'T"
+
+
+
+    [<Test>]
+    let ``Suggest Types in Module`` () =
+        CompilerAssert.TypeCheckSingleError
+            """
+let x : System.Collections.Generic.Lst = ResizeArray()
+
+exit 0
+            """
+            FSharpErrorSeverity.Error
+            39
+            (2, 36, 2, 39)
+            "The type 'Lst' is not defined in 'System.Collections.Generic'\nMaybe you want one of the following:\nList"
+
+
+
+    [<Test>]
+    let ``Suggest Types in Namespace`` () =
+        CompilerAssert.TypeCheckSingleError
+            """
+let x = System.DateTie.MaxValue
+
+exit 0
+            """
+            FSharpErrorSeverity.Error
+            39
+            (2, 16, 2, 23)
+            "The value, constructor, namespace or type 'DateTie' is not defined\nMaybe you want one of the following:\nDateTime\nDateTimeKind\nDateTimeOffset"
+
+
+    //TODO This is identical with the above (only message differ) Suggest Types in Namespace  ???
```

Comment:
This seem a duplicate test

Reply:
I think you can merge them into one using `CompilerAssert.TypeCheckWithErrors` assertion